dspace-api/src/main/java/org/dspace/content/crosswalk/OREIngestionCrosswalk.java

```diff
@@ -50,216 +55,207 @@
+            entryId = ((Attribute) xpathLinks.selectSingleNode(doc)).getValue();
+        } catch (JDOMException e) {
+            throw new CrosswalkException("JDOM exception occurred while ingesting the ORE", e);
         }
-        log.info("OREIngest for Item "+ item.getID() + " took: " + (new Date().getTime() - timeStart.getTime()) + "ms."); 
-	}
-	
-	
-	/**
+
+        // Next for each resource, create a bitstream
+        XPath xpathDesc;
+        NumberFormat nf = NumberFormat.getInstance();
+        nf.setGroupingUsed(false);
+        nf.setMinimumIntegerDigits(4);
+
+        for (Element resource : aggregatedResources) {
+            String href = resource.getAttributeValue("href");
+            log.debug("ORE processing: " + href);
+
+            String bundleName;
+            Element desc = null;
+            try {
+                xpathDesc = XPath.newInstance("/atom:entry/oreatom:triples/rdf:Description[@rdf:about=\"" + this.encodeForURL(href) + "\"][1]");
+                xpathDesc.addNamespace(ATOM_NS);
+                xpathDesc.addNamespace(ORE_ATOM);
+                xpathDesc.addNamespace(RDF_NS);
+                desc = (Element) xpathDesc.selectSingleNode(doc);
+            } catch (JDOMException e) {
+                e.printStackTrace();
+            }
+
+            if (desc != null && desc.getChild("type", RDF_NS).getAttributeValue("resource", RDF_NS).equals(DS_NS.getURI() + "DSpaceBitstream")) {
+                bundleName = desc.getChildText("description", DCTERMS_NS);
+                log.debug("Setting bundle name to: " + bundleName);
+            } else {
+                log.info("Could not obtain bundle name; using 'ORIGINAL'");
+                bundleName = "ORIGINAL";
+            }
+
+            // Bundle names are not unique, so we just pick the first one if there's more than one.
+            List<Bundle> targetBundles = itemService.getBundles(item, bundleName);
+            Bundle targetBundle;
+
+            // if null, create the new bundle and add it in
+            if (targetBundles.size() == 0) {
+                targetBundle = bundleService.create(context, item, bundleName);
+                itemService.addBundle(context, item, targetBundle);
+            } else {
+                targetBundle = targetBundles.get(0);
+            }
+
+            InputStream in = null;
+            if (href != null) {
+                try {
+                    // Make sure the url string escapes all the oddball characters
+                    //String processedURL = encodeForURL(href).replace("%20", "+");
+
+					// Generate a requeset for the aggregated resource
+                    OkHttpClient client = new OkHttpClient();
+                    Request request = new Request.Builder()
+                            .url(href)
+                            .build();
+                    Response response = client.newCall(request).execute();
+                    in = response.body().byteStream();
```

Comment:
**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *in*. Currently, there are execution paths that do not contain closure statements, e.g., when exception is thrown by BitstreamService.create. Either a) close *in* in a try-finally block or b) close the resource by declaring *in* in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).